src/Entities/Player.cpp

```diff
@@ -2235,6 +2284,22 @@ bool cPlayer::LoadFromFile(const AString & a_FileName, cWorldPtr & a_World)
 	m_CurrentXp           = root.get("xpCurrent",      0).asInt();
 	m_IsFlying            = root.get("isflying",       0).asBool();
 
+	Json::Value & JSON_KnownItems = root["knownItems"];
+	for (UInt32 i = 0; i < JSON_KnownItems.size(); i++)
+	{
+		cItem Item;
+		Item.FromJson(JSON_KnownItems[i]);
+		m_KnownItems.insert(Item);
+	}
+
+	auto RecipeNameMap = cRoot::Get()->GetCraftingRecipes()->getRecipeNameMap();
+
+	Json::Value & JSON_KnownRecipes = root["knownRecipes"];
+	for (UInt32 i = 0; i < JSON_KnownRecipes.size(); i++)
+	{
+			m_KnownRecipes.insert(RecipeNameMap[JSON_KnownRecipes[i].asString()]);
```

Comment:
This could fail if the recipe list is changed.
```suggestion
			auto RecipeId = RecipeNameMap.find(JSON_KnownRecipes[i].asString());
			if (RecipeId != RecipeNameMap.end())
			{
				m_KnownRecipes.insert(*RecipeId);
			}
```